### DISCUSSION TITLE
Tiles Directory .metainfo - help needed

### DISCUSSION THREAD

**Discussion by mariush444** [osmandapp/OsmAnd#11864] [Q&A] [answered]
I created the file .metainfo in directory but OSMAND can't download tiles

definition inside .metainfo is 

```
url_template]
https://mapserver.bdl.lasy.gov.pl/arcgis/rest/services/WMTS_BDL_kat_wlasnosci/MapServer/WMTS/tile/1.0.0/WMTS_BDL_kat_wlasnosci/default/GoogleMapsCompatible/{0}/{1}/{2}.png
[ext]
.png
[min_zoom]
6
[max_zoom]
19
[tile_size]
256
[img_density]
16
[avg_img_size]
18000
```

I verified that tiles are accessible. e.g. I can see some tiles like 
https://mapserver.bdl.lasy.gov.pl/arcgis/rest/services/WMTS_BDL_kat_wlasnosci/MapServer/WMTS/tile/1.0.0/WMTS_BDL_kat_wlasnosci/default/GoogleMapsCompatible/16/21509/35937.png

Can anyone find what is wrong?

**Comment by karl-01** [comment, marked as answer]
Try similiar to this:
http://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{$z}/{$y}/{$x}
Especially look at the end

**Reply by mariush444** [reply]
you are right, thank you. x and y are switched.

**Reply by vshcherb** [reply]
I updated documentation but the order is correct, it's z - 0, x - 1, y - 2. But it's changed on server i.e. lat (y-coordinate) / lon (x-coordinate).

**Reply by Bibi56** [reply]
@vshcherb is obviously right. My educated guess - yes a guess - is that depending if the server is using WMS 1.1.1 or WMS 1.3.0, as the bounding box definition changed due to the introduction of the axis order, you may have to switch x and y.
See [QGIS Documentation](https://docs.qgis.org/3.16/en/docs/user_manual/working_with_ogc/ogc_client_support.html#ogc-wms) for hints about "Ignore axis orientation (WMS 1.3/WMTS)".

**Reply by danieldelf** [reply]
@mariush444 I see that you have solved the problem with the property map in osmand, I need help with the "tourist development" program https://mapserver.bdl.lasy.gov.pl/ArcGIS/services/WMS_BDL_Mapa_turystyczna/MapServer/WMSServer what link do I need to provide in osmand to display tourist data?

**Reply by mariush444** [reply]
`https://mapserver.bdl.lasy.gov.pl/ArcGIS/services/WMS_BDL_Mapa_turystyczna/MapServer/WMSServer?FORMAT=image/png8&LAYERS=25,26,27,35&TRANSPARENT=true&SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&STYLES=&SRS=EPSG:4326&BBOX={bbox}&WIDTH=256&HEIGHT=247
`

**Reply by danieldelf** [reply]
@mariush444 Thank you very much, I have one more question - how do I display the layer with the "spend the night in the forest" program?

**Reply by mariush444** [reply]
I have no access to PC now so I can't verify but please add additional layers. I use only 3 categories for my needs but you can use more of them.

**Reply by danieldelf** [reply]
@mariush444 I added layer 76,77 (I deduced this from the url address with the "spend the night in the forest" layer enabled in the browser) but the "spend the night in the forest" layer is not displayed in OSMAND.

**Reply by mariush444** [reply]
Zanocuj seems to be 0

**Reply by mariush444** [reply]
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/8b359400-95d6-4a57-9caf-5d8cae038454"  width="300" height="auto"/> </td>
  </tr>
</table>

**Reply by danieldelf** [reply]
Unfortunately, adding layer 0 still does not display the layer "zanocuj"

**Reply by mariush444** [reply]
Is this orange zone not "zanocuj"?
![IMG_20250505_205957](https://github.com/user-attachments/assets/95d191c5-9345-4cd2-a453-aad38549ee23)

**Reply by danieldelf** [reply]
That's exactly what I mean, please provide the address you enter that displays the "zanocuj" program. When I add layer 0 to the previously provided link, it doesn't display the "zanocuj" areas for me

**Reply by danieldelf** [reply]
I don't know how it happened but now it works, thank you very much for your time and commitment

**Comment by dschneiderch** [comment]
any help getting a link setup to the USFS?
https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer

I tried following a similar url as above but i'm not sure how to interpret this, e.g.
`https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/1/1/1.png` does not work nor does `https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/1/256/256.png` or any other combinations I've tried.

Thanks!

**Reply by mariush444** [reply]
at the first look...  you should use tiles for USA territory because this server supports data for USA only.
Which tile to use you can see here `http://tools.geofabrik.de/map/#12/33.8058/-98.2141&type=Geofabrik_Standard&grid=1`
Your examples are of the scope.

the second - it is about forest departments, so you should find forest maintained by this dept.
`https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/11/780/424.png`

**Reply by dschneiderch** [reply]
OK thanks. I thought the tile row and tilecol would come out of the xml file.
Your example works but shows no data because there is no Forest Service land in that tile.
This one is more obvious https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/13/2879/1457.png

is the .metainfo the same?

Thanks for your help.

**Reply by mariush444** [reply]
this .metainfo below works for me

```
[url_template]
https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/{$z}/{$y}/{$x}.png
[ext]
.png
[min_zoom]
10
[max_zoom]
18
[tile_size]
256
[img_density]
16
[avg_img_size]
18000
```
just create the subdirectory e.g. topo_EDW in directory tiles. Put the .metainfo file in the subdirectory
and that's it

**Reply by jojojones** [reply]
I am trying to get this same online mapsource in to newest Osmand newest IOS/iPhone 12.  Since IOS is sandboxed...Can this be done for the IOS version?

**Reply by mariush444** [reply]
configure map -> map source -> Add (at the end of the list) 
then manually enter the parameters

**Reply by jojojones** [reply]
so close...

Thanks for your response. 

Do I need to create a file “.metainfo” and put it in a directory?…if so that will be hard on an iPhone since iOS is sandboxed unlike android.

If I only need to manually enter the parameters and create an online map link then yes I have done that and I tested and was successful getting FStopo to show up as an overlay on ANDROID using the newest Osmand on BlueStacks emulator using this:

NAME:		any name

URL:		https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/{$z}/{$y}/{$x}.png

MIN:		10

MAX:		18

EXPIRE:		leave blank

MERCATOR:	Pseudo-Mercator projection (default) or Elliptic mercator

SOURCE:	One Image file per tile (default) or SQLiteDB file


I was so excited when it worked…THANK YOU!!!

Then I put the same parameters in iPhone newest Osmand and NOTHING.

???

It appears all of you are on android.

I then tried the same URL with removing the $ signs on the ZYX and some data does overlay but it is not FStopo???

I am less than a mile for the Green Mountain National Forest and FStopo is the only map that appears to have everything.

**Reply by jojojones** [reply]
***EDIT*** I was wrong.  It does work on iOS only with the $ on ZYX removed.  Now the only issue is the zoom.  It only shows zoomed down under 2 miles.  Pretty happy is shows at all.  Now I want to download it and be able to use it offline but not sure how that will work with only a under 2 mile visual range.

**Reply by jojojones** [reply]
do those of you above who have gotten FStopo working using that link...does the FSTopo data layer go away when you zoom out more than a mile?

Here is layer off:
![Screen Shot 2022-01-11 at 4 06 24 PM](https://user-images.githubusercontent.com/60070373/149022200-6746d05c-a6bd-463c-91a7-3ba08d35ff25.png)

layer on:
![Screen Shot 2022-01-11 at 4 06 41 PM](https://user-images.githubusercontent.com/60070373/149022279-aaffc082-d76f-42f8-8a17-550a24975803.png)

zoomed out 2 miles:
![Screen Shot 2022-01-11 at 4 07 00 PM](https://user-images.githubusercontent.com/60070373/149022396-832077e7-8112-4066-80f5-4983f9eada5e.png)

I get the same results with android???

**Reply by mariush444** [reply]
The layer doesn't go away in my opinion.
It is the way how map server presents /shows data.
Can you verify this in on-line web page?
Sorry, I can't find the proper link.

**Comment by jojojones** [comment]
I am using android version of Osmand to generate the picts above.  Any zoom out 2 miles or more changes to the last picture.  The data may still be there in the layer but you can't see any detail.  This happens on both android and iOS using the url cited above.  Here is the same map form Gaia and as you can see all zoom levels show properly.

If you look at the same area on Gaia https://www.gaiagps.com/map/?loc=11.9/-72.9518/43.1249&layer=usfs-2016 it does not change as you zoom out.

please pick any spot and take a grab at 1 mile then 2 miles...does it change to a drab green or does it maintain contour lines and detail.

all using url above with or without $ signs for Android and without $ for iOS.

**Reply by jojojones** [reply]
sorry forgot this overlay only works in national parks...use Winhall, VT as reference.

**Reply by mariush444** [reply]
As I understand osmand works OK and Gaia makes trick that is expected by you but it is wrong way.

Go to any place that you need with this link
http://tools.geofabrik.de/map/#13/43.2136/-72.8379&type=Geofabrik_Standard&grid=1
then you can see what xyz is.

Then use xyz and get FStopo tile needed by you directly from web browser. (remember please, that x&y is in opposite direction in FStopo than in geofabrik)
My example:
https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/10/375/304.png
https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/11/750/609.png
https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/12/1501/1219.png
https://apps.fs.usda.gov/arcx/rest/services/EDW/EDW_FSTopo_01/MapServer/WMTS/tile/1.0.0/EDW_EDW_FSTopo_01/default/GoogleMapsCompatible/13/3003/2438.png

Then you can notice that FSTopo starts to show details from zoom 13 and more.

What Gaia does is that it gets just zoom 13 and use the jpg as a picture.
Then Gaia zoom in and zoom out the picture and not map. Because of that at zoom 18-20 you can see just pixels.
https://www.gaiagps.com/map/?loc=17.1/-72.9698/43.1032&layer=usfs-2016
I don't like it but probably it is what you expect.

What I don't understand is:
I set `zoom min = zoom max = 18` in .metainfo file but osmand works in the same way as for `zoom min=10` and zoom `max=18`.
I would expect that in case `zoom min = zoom max = 18` osmand get tiles only for `zoom 18` and when I want to see something at `zoom=17` or `zoom=19` the osmand will show nothing or start to make the same trick as Gaia. It means zoom in/out tiles of `zoom=18` as a pixels of picture.

Maybe it is bug that zoom in and zoom max is not taken into consideration by osmand and osmand always get everything from map server. This I don't know.

**Reply by jojojones** [reply]
wow...that's fascinating...I am starting to understand more how all this works.

You wrote "Then you can notice that FSTopo starts to show details from zoom 13 and more."  If I look at the other zoom levels less like 12, 11, and 10 then the detail goes away as I have been complaining about.  Since I am looking at these links in a browser and not Osmand and the details go away then isn't this a shortcoming of the tiles or tile server?

Since I understand now (thank you) that I can simply enter the xyz numbers to see a specific tile at a specific zoom level is there a way chose a range?

Also...and this is a big question and I'm starting to think this all is not going to work.  I "thought" Osmand has a way of selecting a  region on the screen of an online map like fstopo or any and downloading it to the phone for offline use but I can't figure it out.???  

I know there are notes online about using mobac and Osmand map create neither of witch I can get running on my Mac or Qgis to make a sqlite file that sources these tiles or actually downloads them then that can be an offline map???....or an online map that parts of which can be downloaded?   I am trying to get a fairly large region of this national forest to display in fstopo on my phone first online then eventually offline.  Gaia mobile and caltopo mobile both can do it by simple download and I think it's all under 3 gigs.

so am I now correct that all this trying to get this online wmts map to display properly will not matter if I am out of cell service offline?

**Reply by mariush444** [reply]
`shortcoming` rather no. It is how author implemented the solution. This is his decision that details started from 13 and not 12.
Just his idea doesn't meet your expectation.

`selecting a region in Osmand` AFAIK there is no such possibility in osmand but the tiles that were seen by you in osmand they are permanently store in osmand for off-line viewing.

You can use MOBAC (probably Qgis too) to select needed region and create maps (sqlite file or set of tiles) for osmand.
I don't know how to put/insert the maps into osmand in IOS. It still possible in Android.

**Reply by vshcherb** [reply]
@mariush444 I think maxzoom should be supported as you said, so it shouldn't get from server if it's in configuration and it should overscale at least for zoom + 2. Please open the bug if it's not the case

**Reply by jojojones** [reply]
thank you both for your help.

**Reply by mariush444** [reply]
`selecting a region in Osmand` I was wrong.  
Someone in Telegram found that it was possible https://docs.osmand.net/en/main@latest/osmand/map/raster-maps#download--update-tiles
So you can grab needed region using osmand.

**Comment by jojojones** [comment]
thanks...I feel pretty dopey now.  I have actually seen this maybe a year ago but I have been trying too many other programs and forgot it can do it.  THANKS.  I will test with the FSTopo I did get running even though it's zoom was limited.
